### ISSUE TITLE
How can I improve the efficiency when I draw many polylines on map?

### ISSUE THREAD

**Issue by zhaoxuhui** [osmdroid/osmdroid#485]
Now I need to load many polylines from shapefile and draw them on the map.Such as the road network in a city.The problem is that when I draw these polylines using:
```
mMapView.getOverlays().add(polyline);
mMapView.invalidate();
```
The map will become very slow to move and zoom.So I want to solve this problem.Can you offer some ideas for this?Thanks~

**Comment by spyhunter99** [comment]
This gets asked once a month or so. Basically, reduce the number of points on the poly lines using the douglas pucker algorithm and then dynamically readjust as needed when the map zooms in and out. Also, you can clip the polyline at the bounds of the screen, then as the user pans, rerender the lines based on the new extent. Lines that are completely out of the view can then be removed from the overlay manager. Using these mechanisms, you can get pretty good performance with a lot of overlays

**Comment by zhaoxuhui** [comment]
OK.I'll try your methods and comment again if I have any other problems.Thanks~

**Comment by spyhunter99** [comment]
see also #353 there's also been a number of discussions in other tickets on this. Probably worth spending some time to dig them up. I think I covered most of the points in the previous comment though

**Comment by monsieurtanuki** [comment]
@zhaoxuhui I guess the polylines performances should be better with #725, that automatically clips the polygons and polylines.
If it's still not fast enough, there are still improvements I can implement (on the clip itself), so let me know.

**Comment by spyhunter99** [comment]
see the branch #906. it probably won't be merged until 6.1.0 after sufficient testing, but it uses the bounds of the overlay to first check on whether or not to draw it.
